modules/ROOT/pages/monitoring/logging.adoc

```diff
@@ -1,215 +1,334 @@
-        </RollingFile>
+            <DefaultRolloverStrategy fileIndex="min" max="7"/>
+        </RollingRandomAccessFile>
     </Appenders>
 
-    <Loggers>
-        <Root level="INFO">
-            <AppenderRef ref="DebugLog"/>
+    <Loggers> \ #<9>
+        <!-- Log levels. One of DEBUG, INFO, WARN, ERROR or OFF -->
+
+        <!-- The debug log is used as the root logger to catch everything -->
+        <Root level="INFO"> \ #<10>
+            <AppenderRef ref="DebugLog"/> <!-- Keep this -->
         </Root>
 
+        <!-- The query log, must be named "QueryLogger" -->
+        <Logger name="QueryLogger" level="INFO" additivity="false"> \ #<11>
+            <AppenderRef ref="QueryLog"/>
+        </Logger>
+
+        <!-- The http request log, must be named "HttpLogger" -->
         <Logger name="HttpLogger" level="INFO" additivity="false">
             <AppenderRef ref="HttpLog"/>
         </Logger>
+
+        <!-- The security log, must be named "SecurityLogger" -->
+        <Logger name="SecurityLogger" level="INFO" additivity="false">
+            <AppenderRef ref="SecurityLog"/>
+        </Logger>
     </Loggers>
 </Configuration>
 ----
+======
+=====
 
-
-.Breakdown of the illustrative Log4j 2 configuration file line per line
-[cols="8h,~", options="header"]
-|===
-| Line(s)
-| Description
-
-| 1
-| Configuration tag with a `monitorInterval` of 30 seconds and a package namespace of `org.neo4j.logging.log4j`. +
+<1> Configuration tag with a `monitorInterval` of 30 seconds and a package namespace of `org.neo4j.logging.log4j`. +
 The monitor interval tells Log4j to periodically check the XML file for changes and reload the file if a change is detected. +
 The package namespace gives access to the Neo4j configuration lookup with `${config:<setting>}`.
 
-| 3, 13
-| Defines two `<RollingFile>` appenders.
-  One writes to _debug.log_ and one to _http.log_.
-  The `name` must be unique and is used later when referencing the appender.
+<2> Appenders are used to define where the log messages are written.
+The `name` of the appender must be unique as it is used by the logger when referencing the appender.
+Appenders write to _debug.log_, _http.log_, _query.log_, and _security.log_.
 
-| 4, 14
-| Specifies a `filePattern` to be used when the file is rolled.
-  The pattern renames the files to _debug.log.01_ and _http.log.01_ when they reach the defined trigger.
+<3> By default, Neo4j uses the `<RollingRandomAccessFile>` appender as it shows a performance increase between 20-200% as per the Log4j documentation.
+However, all writes are buffered, which means that if the server crashes, the last log messages might be lost.
+If that is not acceptable for you, use the `<RollingFile>` appender instead.
+See <<general-logging-appenders>> for more information.
 
-| 5
-| Defines the layout to use for the `DebugLog` appender, in this case with the `GMT+2` timezone.
+<4> `filePattern` specifies a file pattern to be used when the file is rolled.
```

Comment:
For items 4 and 5 I would add a link to log4j docs since they are referencing log4j specific attributes in the body of patterns

Reply:
I added a link to the section where there is also a link to the log4j docs.